pkg/doc/jsonld/context/importer.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package context
+
+import (
+	"bufio"
+	"embed"
+	"encoding/json"
+	"fmt"
+
+	"github.com/piprate/json-gold/ld"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/spi/storage"
+)
+
+const (
+	DefaultContextDBName = "ContextsDB"
+)
+
+var logger = log.New("aries-framework/jsonld/context")
+
+// Importer imports JSON-LD context documents into the underlying storage.
+type Importer struct {
+	store storage.Store
+}
+
+type importerOpts struct {
+	contextDBName   string
+	preloadContexts bool
+}
+
+// ImporterOpts configures Importer during creation.
+type ImporterOpts func(opts *importerOpts)
+
+// NewImporter returns a new Importer instance.
+func NewImporter(storageProvider storage.Provider, opts ...ImporterOpts) (*Importer, error) {
+	importerOpts := &importerOpts{
+		contextDBName: DefaultContextDBName,
+	}
+
+	for i := range opts {
+		opts[i](importerOpts)
+	}
+
+	store, err := storageProvider.OpenStore(importerOpts.contextDBName)
+	if err != nil {
+		return nil, fmt.Errorf("open store for context DB: %w", err)
+	}
+
+	if importerOpts.preloadContexts {
+		if err := importContexts(store); err != nil {
+			return nil, fmt.Errorf("import default contexts: %w", err)
+		}
+	}
+
+	return &Importer{store: store}, nil
+}
+
+func importContexts(store storage.Store) error {
+	var ops []storage.Operation
+
+	for u := range contexts {
+		c, err := readContextDocument(contexts[u].path)
+		if err != nil {
+			return fmt.Errorf("read context document: %w", err)
+		}
+
+		remoteDocument := ld.RemoteDocument{
+			DocumentURL: contexts[u].documentURL,
+			Document:    c,
+		}
+
+		b, err := json.Marshal(remoteDocument)
+		if err != nil {
+			return fmt.Errorf("marshal remote document: %w", err)
+		}
+
+		ops = append(ops, storage.Operation{Key: u, Value: b})
+	}
+
+	if err := store.Batch(ops); err != nil {
```

Comment:
I assume if a context is already loaded into the DB then it is overwritten?

Note: We will have new contexts to load overtime so I think importer will logically be run on a fresh storage AND on version upgrades. We will need to add missing or updated contexts on these upgrades.

Reply:
Note that `Put` or `Batch` will overwrite data.